admin/build.sh

```diff
@@ -9,45 +9,44 @@ set -e
-# `.travis.yml`, otherwise the required build doesn't run.
+# In order to build a previously released version against a new (binary incompatible) Scala release,
+# a new commit that modifies (and prunes) the Scala versions in .travis.yml needs to be added on top
+# of the existing tag. Then a new tag can be created for that commit, e.g., `v1.2.3#2.13.0-M5`.
+# Everything after the `#` in the tag name is ignored.
+
+if [[ "$TRAVIS_JDK_VERSION" == "openjdk6" && "$TRAVIS_SCALA_VERSION" =~ 2\.11\..* || "$TRAVIS_JDK_VERSION" == "oraclejdk8" && "$TRAVIS_SCALA_VERSION" =~ 2\.1[23]\..* ]]; then
+  RELEASE_COMBO=true;
+fi
+
+if [ "$SCALAJS_VERSION" = "" ]; then
+  projectPrefix="xml"
+else
+  projectPrefix="xmlJS"
+fi
 
 verPat="[0-9]+\.[0-9]+\.[0-9]+(-[A-Za-z0-9-]+)?"
-tagPat="^v$verPat(#$verPat#[0-9]+)?$"
+tagPat="^v$verPat(#.*)?$"
 
 if [[ "$TRAVIS_TAG" =~ $tagPat ]]; then
-  currentJvmVer=$(java -version 2>&1 | awk -F '"' '/version/ {print $2}' | sed 's/^1\.//' | sed 's/[^0-9].*//')
-
   tagVer=$(echo $TRAVIS_TAG | sed s/#.*// | sed s/^v//)
   publishVersion='set every version := "'$tagVer'"'
 
-  scalaAndJvmVer=$(echo $TRAVIS_TAG | sed s/[^#]*// | sed s/^#//)
-  if [ "$scalaAndJvmVer" != "" ]; then
-    scalaVer=$(echo $scalaAndJvmVer | sed s/#.*//)
-    jvmVer=$(echo $scalaAndJvmVer | sed s/[^#]*// | sed s/^#//)
-    if [ "$jvmVer" != "$currentJvmVer" ]; then
-      echo "Not publishing $TRAVIS_TAG on Java version $currentJvmVer."
-      exit 0
-    fi
-    publishScalaVersion='set every ScalaModulePlugin.scalaVersionsByJvm := Map('$jvmVer' -> List("'$scalaVer'" -> true))'
-    echo "Releasing $tagVer using Scala $scalaVer on Java version $jvmVer."
-  else
-    echo "Releasing $tagVer on Java version $currentJvmVer according to 'scalaVersionsByJvm' in build.sbt."
-  fi
+  if [ "$RELEASE_COMBO" = "true" ]; then
+    currentJvmVer=$(java -version 2>&1 | awk -F '"' '/version/ {print $2}' | sed 's/^1\.//' | sed 's/[^0-9].*//')
+    echo "Releasing $tagVer with Scala $TRAVIS_SCALA_VERSION on Java version $currentJvmVer."
 
-  extraTarget="+publish-signed"
-  cat admin/gpg.sbt >> project/plugins.sbt
-  cp admin/publish-settings.sbt .
+    publishTask="$projectPrefix/publish-signed"
 
-  # Copied from the output of genKeyPair.sh
-  K=$encrypted_6b8d67feaab7_key
-  IV=$encrypted_6b8d67feaab7_iv
+    cat admin/gpg.sbt >> project/plugins.sbt
+    cp admin/publish-settings.sbt .
 
-  openssl aes-256-cbc -K $K -iv $IV -in admin/secring.asc.enc -out admin/secring.asc -d
+    # Copied from the output of genKeyPair.sh
+    K=$encrypted_6b8d67feaab7_key
+    IV=$encrypted_6b8d67feaab7_iv
+    openssl aes-256-cbc -K $K -iv $IV -in admin/secring.asc.enc -out admin/secring.asc -d
+  fi
 fi
 
-sbt "$publishVersion" "$publishScalaVersion" clean update +test +publishLocal $extraTarget
+sbt "++$TRAVIS_SCALA_VERSION" "$publishVersion" "$projectPrefix/clean" "$projectPrefix/test" "$projectPrefix/publishLocal" "$publishTask"
```

Comment:
Sorry to see the `update` task go.

I don't know if `"$projectPrefix/clean"` is a worthwhile distinction.

But these are nits.

Reply:
`update` runs anways as part of `test`

Reply:
Sure, I like the timestamp and success of each task I guess.